lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
Just a side note, `new Uint8Array(-1)` also throws with close to the same error message. If you're calling that directly then can possibly forego validating the range (if it's going to throw).